moveit_ros/moveit_servo/src/collision_monitor.cpp

```diff
@@ -95,8 +95,8 @@ void CollisionMonitor::checkCollisions()
     const double self_velocity_scale_coefficient{ log_val / servo_params_.self_collision_proximity_threshold };
     const double scene_velocity_scale_coefficient{ log_val / servo_params_.scene_collision_proximity_threshold };
 
-    // Reset the scale on every iteration.
-    collision_velocity_scale_ = 1.0;
+    // Use a local variable to prevent access issues between the collision monitor and the servo loop
+    auto new_collision_velocity_scale = 1.0;
 
     if (servo_params_.check_collisions)
```

Comment:
Another alternative is to do away entirely with the local variable, and simply set the collision scale to 0.0 in the `else` of this statement. Either/or

Reply:
Yeah I think it's 6 of one, half dozen of the other. 

In this instance I'll lean towards the extra variable to make in painfully obvious that it's a shared variable and it shouldn't have intermediate values set while it's being computed. It wasn't immediately obvious to me how `collision_velocity_scale_` works until you explained it to me, and this way I think it's more clear to new devs working on servo.

Reply:
In that spirit I updated the comment to make it even more obvious...

Reply:
Ignore my previous comments here it is without a local variable.